presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -167,6 +167,23 @@ public Response createQuery(
                     .build());
         }
 
+        // The Teradata Presto ODBC Driver checks node version to decide the Presto's statement
+        // protocol and sends test queries about PREPARE statement.
+        // Rewrite the statement so that Presto always returns version for the compatible protocol.
+        // Ban the statement which will never be consumed by the driver.
+        if (servletRequest.getHeader("User-Agent").equals("Teradata Presto ODBC Driver")) {
+            if (statement.equals("select node_version from system.runtime.nodes where coordinator=true")) {
```

Comment:
For this fix, we purely want to interpret those pre-compiled queries. We shouldn't consider mixed-case for the statement because the mixed-case statement is indicating it must not be submitted by the driver.